### PR TITLE
Refactor curl_to_request_kwargs #5001

### LINKED ISSUE #5001: refactoring curl_to_request_kwargs to reduce cyclomatic complexity
## Summary

After some exploring with cyclomatic complexity tools (lizard), the function was found to have the second highest complexity. 

## Motivation

Low complexity allows for higher readability, testability and maintainability. 

## Solution

Refactor

## Additional context

N/A

### PR DESCRIPTION
**Pull request by djikoSal**
Co-authored-by: alkazaz alkazaz@kth.se
Co-authored-by: swill swill@kth.se
Co-authored-by: lerjevik lerjevik@kth.se
Co-authored-by: aljica aljica@kth.se

Resolves #5001

### CODE REVIEW

**Comment by djikoSal** [comment]
before: CCN 15
after: CCN 5

File: scrapy/utils/curl.py
Comment (Gallaecio) [inline scrapy/utils/curl.py:36]: Please, use `_` as a prefix for functions like these that are not meant to be imported by users.

**Review by Gallaecio — COMMENTED** [review]
I find the refactoring a bit harder to read than before. I wonder if this can be done differently.

For example, I think it would make sense to extract the code to parse headers and cookies into a `_parse_headers_and_cookies` function. And if that is enough to give the current function a more normal cyclomatic complexity, I think we should leave it at that, and not make further changes.

**Comment by djikoSal** [comment]
Thanks for your feedback! We have now changed the refactoring to a new func _parse_headers_and_cookies. @Gallaecio

File: scrapy/utils/curl.py
Comment (Gallaecio) [inline scrapy/utils/curl.py:91]:
```suggestion
    headers, cookies = _parse_headers_and_cookies(parsed_args)
```

Comment (djikoSal) [reply scrapy/utils/curl.py]: Whoops, sorry about that. Fixed now!

**Comment by Gallaecio** [comment]
[Flake8 reports a few style issues](https://github.com/scrapy/scrapy/pull/5002/checks?check_run_id=1968542941):

```
/home/runner/work/scrapy/scrapy/scrapy/utils/curl.py:36:1: E302 expected 2 blank lines, found 1
/home/runner/work/scrapy/scrapy/scrapy/utils/curl.py:48:1: W293 blank line contains whitespace
/home/runner/work/scrapy/scrapy/scrapy/utils/curl.py:55:1: E302 expected 2 blank lines, found 1
```

**Comment by Gallaecio** [comment]
A tip for the future: squashing commits and force-pushing during a code review makes it slightly more difficult for reviewers to see what changed since they last reviewed the code. It’s usually better to just add additional commits. We maintainers have a button to squash your commits before merging, so if your concern is commit history pollution, you should not worry about that.

**Comment by djikoSal** [comment]
Thank you for the tip and thanks for the patience! This is our first contribution, so we are still learning the ins and outs of contributing correctly 😅 

There seems to be a [failing test](https://github.com/scrapy/scrapy/pull/5002/checks?check_run_id=1969058333) still (macOS / tests 3.7), I am not sure of the cause.

**Comment by Gallaecio** [comment]
We have random tests in Mac OS X, I’ll keep re-running it until it passes, but you can assume it will pass.

**Review by Gallaecio — APPROVED** [review]

**Comment by Gallaecio** [comment]
Thanks!

**Pull request merged**
